sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -105,7 +96,14 @@ case class CreateViewCommand(
     }
     val sessionState = sparkSession.sessionState
 
-    if (isTemporary) {
+    // 1) CREATE VIEW: create a temp view when users explicitly specify the keyword TEMPORARY;
+    //                 otherwise, create a permanent view no matter whether the temporary view
+    //                 with the same name exists or not.
+    // 2) ALTER VIEW: alter the temporary view if the temp view exists; otherwise, try to alter
+    //                the permanent view. Here, it follows the same resolution like DROP VIEW,
+    //                since users are unable to specify the keyword TEMPORARY.
+    if (viewType == ViewType.Temporary ||
+        (viewType != ViewType.Permanent && sessionState.catalog.isTemporaryTable(name))) {
```

Comment:
This is not so readable, how about we use 3 branches for temp view and permanent view and any view individually?

Also, we don't need to mention CREATE VIEW or ALTER VIEW here, the semantic is clearly defined by the SaveMode and ViewType, we just need to document how these codes match the semantic.